reconcile/ocm_upgrade_scheduler.py

```diff
@@ -67,72 +70,59 @@ def fetch_desired_state(
     return sorted_desired_state
 
 
-def update_history(history, upgrade_policies):
-    """Update history with information from clusters
-    with upgrade policies.
+def update_history(version_data: VersionData, upgrade_policies: list[dict[str, Any]]):
+    """Update history with information from clusters with upgrade policies.
 
     Args:
-        history (dict): history in the following format:
-        {
-          "check_in": "2021-08-29 18:01:27.730441",
-          "versions": {
-            "version1": {
-                "workloads": {
-                    "workload1": {
-                        "soak_days": 21,
-                        "reporting": [
-                            "cluster1",
-                            "cluster2"
-                        ]
-                    },
-                        "workload2": {
-                        "soak_days": 6,
-                        "reporting": [
-                            "cluster3"
-                        ]
-                    }
-                }
-            }
-          }
-        }
+        history (VersionData): version data, including history of soakdays
         upgrade_policies (list): query results of clusters upgrade policies
     """
-    default_workload_history = {
-        "soak_days": 0.0,
-        "reporting": [],
-    }
-
     now = datetime.utcnow()
-    check_in = parser.parse(get_or_init(history, "check_in", str(now)))
-    versions = get_or_init(history, "versions", {})
+    check_in = parser.parse(version_data.check_in or str(now))
```

Comment:
can you explain the wording `check_in` in this context?

Reply:
this is from the previous implementation. It is the date/time of the last data update from the integration.